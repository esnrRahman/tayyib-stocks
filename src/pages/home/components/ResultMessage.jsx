import React from 'react';
import { Message } from 'semantic-ui-react';
import PropTypes from 'prop-types';

import { StyledMessageHeader } from './ResultMessageStyles';

const ResultMessage = ({ size, title, description, isFailure }) => (
  <Message size={size || 'large'} positive={!isFailure} negative={isFailure}>
    <StyledMessageHeader>
      {title}
    </StyledMessageHeader>
    <p>
      {description}
    </p>
  </Message>
);

ResultMessage.propTypes = {
  size: PropTypes.string,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  isFailure: PropTypes.bool.isRequired,
};

export default ResultMessage;
