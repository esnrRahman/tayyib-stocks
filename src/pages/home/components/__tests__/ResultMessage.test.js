import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ResultMessage from '../ResultMessage';

describe('<ResultMessage />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      title: 'Test title',
      description: 'Test description',
      isFailure: false,
    };
  });

  it('renders with default attributes', () => {
    const wrapper = shallow(
      <ResultMessage
        {...initialProps}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with custom size', () => {
    const customProps = { ...initialProps, size: 'small' };
    const wrapper = shallow(<ResultMessage {...customProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with isFailure prop set to true', () => {
    const customProps = { ...initialProps, isFailure: true };
    const wrapper = shallow(<ResultMessage {...customProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
