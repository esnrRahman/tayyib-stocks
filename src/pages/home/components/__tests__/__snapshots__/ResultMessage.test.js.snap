// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultMessage /> renders with custom size 1`] = `
<Message
  negative={false}
  positive={true}
  size="small"
>
  <Styled(MessageHeader)>
    Test title
  </Styled(MessageHeader)>
  <p>
    Test description
  </p>
</Message>
`;

exports[`<ResultMessage /> renders with default attributes 1`] = `
<Message
  negative={false}
  positive={true}
  size="large"
>
  <Styled(MessageHeader)>
    Test title
  </Styled(MessageHeader)>
  <p>
    Test description
  </p>
</Message>
`;

exports[`<ResultMessage /> renders with isFailure prop set to true 1`] = `
<Message
  negative={true}
  positive={false}
  size="large"
>
  <Styled(MessageHeader)>
    Test title
  </Styled(MessageHeader)>
  <p>
    Test description
  </p>
</Message>
`;
